mlflow/models/evaluation/default_evaluator.py

```diff
@@ -369,6 +386,26 @@ def __validate_artifacts(artifacts):
     )
 
 
+def _compute_df_mode_or_mean(df):
+    """
+    Compute mean (for continuous columns) and compute mode (for other columns) for the
+    input dataframe, return a dict, key is column name, value is the corresponding mode or
+    mean value, this function calls `_is_continuous` to determine whether the
+    column is continuous column.
+    """
+    df_cols = df.columns.tolist()
+    continuous_cols = [c for c in df_cols if _is_continuous(df[c])]
+    means = df[continuous_cols].mean().to_dict()
+    if continuous_cols == df_cols:
+        return means
+    else:
+        modes = df.drop(continuous_cols, axis=1).mode().loc[0].to_dict()
+        return {**means, **modes}
```

Comment:
`df[continuous_cols]` could be empty. I would do something like this:

```suggestion
    a = df[continuous_cols]
    b = df.drop(continuous_cols, axis=1)
    
    means = {} if a.empty else a.mean().to_dict()
    modes = {} if b.empty else b.loc[0].mean().to_dict()
    return {**means, **modes}
```

Reply:
or you can use:

```
df.mode().head(1).squeeze().to_dict()
```

this works even if `df` is empty.